Release 0.11.0

akaila-splunk opened the pull request:
- Update all dependencies to latest stable version and replace deprecated dependencies
- make library compatible with node version 10.0 and 14.0 and Splunk version 8.0 and 8.2.0
- Configured GitHub action for test and release
- Update library to use 'make up' and 'make test' convention

fantavlik commented on splunklogger.js:
```diff
@@ -14,7 +14,7 @@
  */
 
-var request = require("request");
+var needle = require("needle");
```
- [ ] Let's add an item to the CHANGELOG.md for this - under `New features & APIs`: Replaced client implementation to use `needle` in lieu of deprecated `request`

akaila-splunk replied:
Updated CHANGELOG.md file as suggested.

fantavlik commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,20 @@
+name: Create Release
```
- [ ] Let's add an item to the CHANGELOG.md for this - under `Minor changes`: Converted CI from Travis to Github Actions and added CD

akaila-splunk replied:
Updated CHANGELOG.md file as suggested.

fantavlik commented on .travis.yml:
```diff
@@ -10,7 +10,7 @@ before_install:
+    - echo password=changed! >> $HOME/.splunkrc
     # Set SPLUNK_HOME
     - export SPLUNK_HOME="/opt/splunk"
     # Pull docker image
```
Can we remove the .travis.yml file actually? Feel free to do that in this PR

akaila-splunk replied:
Removed .travis.yml flie.

fantavlik reviewed (changes requested):
Looking good so far, I left comments in the review but two things to take care of:

- [ ] Need to add a new section to the CHANGELOG.md for this new release (notes on what to add to various sections are in the review below)
- [ ] Let's remove the .travis.yml since that's no longer used

akaila-splunk commented:
> Looking good so far, I left comments in the review but two things to take care of:
> 
> * [ ]  Need to add a new section to the CHANGELOG.md for this new release (notes on what to add to various sections are in the review below)
> * [ ]  Let's remove the .travis.yml since that's no longer used

Updated CHANGELOG.md file as suggested and removed .travis.yml file.

fantavlik reviewed (approved):
Looking good, thanks again!
